src/bindings/node/node.cc

```diff
@@ -0,0 +1,236 @@
+#include <node_api.h>
+
+#include <boost/make_shared.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/shared_ptr.hpp>
+#include <sstream>
+#include <string>
+#include <iostream>
+
+#include "valhalla/tyr/actor.h"
+
+//NOTE: the Route function has a lot of boiler plate in it so it would be good to break that out
+//so that it can be re-used for all of the various functions that need to do input output marshalling
+
+#define DECLARE_NAPI_METHOD(name, func) { name, 0, func, 0, 0, 0, napi_default, 0 }
+
+class Actor {
+ public:
+  static napi_value Init(napi_env env, napi_value exports) {
+    napi_property_descriptor properties[] = {
+        DECLARE_NAPI_METHOD("route", Route)
+        //many more...
+    };
+
+    napi_value cons;
+    napi_status status =
+        napi_define_class(env, "Actor", NAPI_AUTO_LENGTH, New, nullptr, 1, properties, &cons);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to define Actor class");
+    }
+
+    status = napi_create_reference(env, cons, 1, &constructor);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to create reference to constructor");
+    }
+
+    status = napi_set_named_property(env, exports, "Actor", cons);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to set Actor property on exports");
+    }
+
+    status = napi_create_reference(env, cons, 1, &constructor);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to create reference to constructor");
+    }
+
+    status = napi_create_function(env, "configure", NAPI_AUTO_LENGTH, Configure, nullptr, &exports);
+    if (status != napi_ok) {
+      napi_throw_error(env, NULL, "Failed to create function from Configure");
+    }
+
+    return exports;
+  }
+
+  static void Destructor(napi_env env, void* nativeObject, void* finalize_hint) {
+    // reinterpret_cast<valhalla::tyr::actor_t*>(nativeObject)->~MyObject();
+    delete reinterpret_cast<valhalla::tyr::actor_t*>(nativeObject);
```

Comment:
this shouldnt be `valhalla::tyr::actor_t*` it should be `Actor`. in other words its the wrapper object not the object that it encapsulates